tests/structure_test.js

```diff
@@ -1039,7 +1059,7 @@ describe('structure', function () {
         ref.observe(function () { i++; });
         ref2.observe(function () { i2++; });
         ref2.cursor().update(function () { return 'updated'; });
-        ref2.observe(function () { if(i == 1 && i2 == 2) done(); });
+        ref2.observe(function () { if(i == 1 && i2 == 1) done(); });
```

Comment:
Sorry, I've missed the last line of the test :-). As I've used Set as a container for listener methods, the order was not guaranteed and callback with done was called before i2++. Set was changed to OrderedSet